contracts/oracle/PCVOracle.sol

```diff
@@ -306,47 +272,46 @@ contract PCVOracle is IPCVOracle, CoreRefV2 {
         emit VenueOracleUpdated(venue, oldOracle, newOracle);
     }
 
-    function _getUsdPcvDelta(
+    function _updateBalance(
         address venue,
-        int256 pcvDelta
-    ) private view returns (int256) {
+        int256 deltaBalance,
+        int256 deltaProfit,
+        bool isLiquid
+    ) private {
+        // Read oracle to get USD values of delta
         address oracle = venueToOracle[venue];
         require(oracle != address(0), "PCVOracle: invalid caller deposit");
         (uint256 oracleValue, bool oracleValid) = IOracleV2(oracle).read();
         require(oracleValid, "PCVOracle: invalid oracle value");
-        return (int256(oracleValue) * pcvDelta) / 1e18;
-    }
-
-    function _afterActionHook() private {
-        if (voltOracle != address(0)) {
-            DynamicVoltSystemOracle(voltOracle).updateActualRate(
-                lastLiquidVenuePercentage()
-            );
-        }
-    }
-
-    function _updateBalance(int256 pcvDeltaUSD, bool isLiquid) private {
-        uint256 oldLiquidity = isLiquid
-            ? lastLiquidBalance
-            : lastIlliquidBalance;
-
-        uint256 newLiquidity;
-        if (pcvDeltaUSD < 0) {
-            newLiquidity = oldLiquidity - (pcvDeltaUSD * -1).toUint256();
-        } else {
-            newLiquidity = oldLiquidity + pcvDeltaUSD.toUint256();
-        }
-
-        if (isLiquid) lastLiquidBalance = newLiquidity;
-        else lastIlliquidBalance = newLiquidity;
+        // Compute USD values of delta
+        int256 deltaBalanceUSD = (int256(oracleValue) * deltaBalance) / 1e18;
+        int256 deltaProfitUSD = (int256(oracleValue) * deltaProfit) / 1e18;
 
+        // Emit event
         emit PCVUpdated(
             msg.sender,
             isLiquid,
             block.timestamp,
-            oldLiquidity,
-            newLiquidity
+            deltaBalanceUSD,
+            deltaProfitUSD
         );
+
+        // SLOAD accumulative profits of the venue
+        int256 oldProfits = lastVenueProfits[venue];
+        int256 newProfits = oldProfits + deltaProfitUSD;
+        // SSTORE accumulative profits of the venue, if different,
+        // and emit event.
+        if (oldProfits != newProfits) {
```

Comment:
couldn't this check be simplified to `deltaProfitUSD != 0` ?

Reply:
then you can remove the `newProfits` variable, and on line 305, you can do 

`lastVenueProfits[venue] += deltaProfitsUSD;`